### ISSUE TITLE
Compare with dialyzer

### ISSUE THREAD

**Issue by fenollp** [josefs/Gradualizer#119]
Interesting project! It’s great to see more tooling around the BEAM :)

I just came about 
https://github.com/erlang/otp/pull/2069 and couldn’t find documentation expressing the differences between the 2 projects as apparently there are some: gradualizer finds different things than dialyzer!
But to what extent? How does this work?
I’d love to see at least a section in the readme about this. 

Thanks!

**Comment by zuiderkwast** [comment]
Documentation is on the TODO list. :-) There is some draft documentation in the wiki of this project.

In short, Gradualizer is based on the theory of *gradual typing*, which gives certain guarantees under certain circumstances. Dialyzer has some other theory. Dialyzer is described as a *discrepancy analyzer* rather than a type checker.

**Comment by josefs** [comment]
Indeed, we should populate the wiki with info about this. In the meantime you can have a look at a talk I gave before summer at Code Beam:
https://www.youtube.com/watch?v=_u1NDuFsW2A
There's a comparison with Dialyzer that starts at 33:50.

**Comment by uabboli** [comment]
Thank you!
